### ISSUE TITLE
Never backups 

### ISSUE THREAD

**Issue by ainquisitive** [meganz/MEGAcmd#240]
Hello All, 

I'm trying to back up my NAS **_(Synology)_** to the mega cloud via MEGACMD and I followed the guide provided by mega and on GitHub but it doesn't work properly. it only copies folder, not the content. I tried so many times but no luck.  Is there anyone has any script written already (example) ..? I would like to BACKUP first and then sync whenever there are changes.

Many Thank
**Mahi**

**Comment by ainquisitive** [comment]
The steps I have taken. 

Installed MEGACMD in Synology NAS VIA Synology package centre
Access Synology Via Putty
Script 
!bin/bash
#mega-login anyemail@domain.com Password*
mega-put /volume
mega-sync /volume
mega-sync
Task added to the task scheduler in NAS to run the script. 
 After this whenever I checked mega cloud account it only shows me folders but not the data inside the folder or folder structure only.  Synology NAS, I'm not getting any errors.

**Comment by mattw-mega** [comment]
Hi, all you need to achieve that is to first set up the sync, one time: `mega-sync /volume /cloudfolder`  (specify the folder in the cloud you want /volume synced to).   Then, so long as `mega-cmd-server` is running, it will sync those (local changes applied to the cloud, and cloud changes applied locally (please do make sure that is what you want)).  thanks.

**Comment by ainquisitive** [comment]
Hi Mega, We can sync alright, but no files get synced. Just the folders, looking into the command, it seems it's the intention to do this, hence the put - so should we be syncing and then pushing?

**Comment by polmr** [comment]
`mega-put /volume`  -> This would upload the folder /volume. This will be finished only after volume is uploaded
`mega-sync /volume` -> this would fail (no remote folder added)
`mega-sync` -> this will simple list the syncs configured. 

You don't need to do `mega-put`, once you configure a sync, the sync engine should initiate the transfers and folders creation for you. You can see the list and progress of transfers with `mega-transfers --show-syncs`.  Please, ensure that `mega-sync` shows your transfer configured correctly and the state is `Active`

**Comment by ainquisitive** [comment]
Hey,

Thanks for this, but we initially used just sync and it only does folders. After looking at docs, can see that this is intentional behaviour. We would like the files too, hence the -put idea

how do we get folders and files onto mega from NAS via bash script?

Thanks

**Comment by polmr** [comment]
> After looking at docs, can see that this is intentional behaviour

sync does sync folders and files. It works the same way as MEGAsync does. 

> how do we get folders and files onto mega from NAS via bash script?

If you want to upload a folder use `mega-put /local/folder /remote/location`. If you want to synchronize a local folder with a folder in MEGA, then stablish a sync (once, you don't need to execute it everytime. If mega-cmd-server is running, whenever you add files/folders locally, they would be uploaded to MEGA). Bear in mind that sync is two-way sync: files deleted remotely will be removed locally. 

Further reading:
https://github.com/meganz/MEGAcmd
https://github.com/meganz/MEGAcmd/blob/master/UserGuide.md
https://mega.nz/help/client/megasync/general/how-does-syncing-with-mega-work-576c7fae886688e7028b4580

**Comment by ainquisitive** [comment]
Hi, 

as mentioned before, we initially used sync, but it only did folders. In the documentation there is no mention of files until after the folders are created. We tried using put to upload the files before syncing them, but that also didn't work. We managed to get folders 100%, but there are no files inside of them.

**Comment by polmr** [comment]
Folder upload works recursively, it creates folders and initiates file transfers once these folders are ready. It seems as if your file transfers was on hold for some reason. 
You can see the list and progress of transfers with `mega-transfers --show-syncs`. Please, ensure that mega-sync shows your sync configured correctly and the state is `Active`. 
If that doesn't solve your issue, and you can reproduce it, it might be interesting to have a log of the server while that happens. In order to do so.
Please, close all megacmd instances and execute the server in a separate console:
Open a console and type
```
killall -2 mega-cmd-server
mega-cmd-server --debug-full | tee megacmd.log
```
Then, open another console and then try to reproduce the issue. Finally quit the server (Ctrl+C in the console that run it should do).
Then please send us that `megacmd.log` file. The log might contain some sensitive data. Feel free to remove anything you don't want to share with us.
If you do not want to post it here (you'd rather not), send it to support@mega.nz and refer this issue in your email so that it can reach development team. Also, please include the outputs of `mega-sync` & `mega-transfers --show-syncs` commands.

**Comment by ainquisitive** [comment]
Hi @polmr 

Tried as you advised 
this is what happened:
Doing:
kill all -2 mega-cmd-server
it returns:
> mega-cmd-server: no process found
then I do:
mega-cmd-server --debug-full | tee megacmd.log
it returns:
> Another instance of MEGAcmd Server is running. Execute with --skip-lock-check to force running (NOT RECOMMENDED)

I can't generate the log :(

**Comment by polmr** [comment]
`kill all` should read `killall`
if `-2` fails, do `-15` afterwards
Ensure `pgrep -a mega-cmd-server` returns nothing

**Comment by polmr** [comment]
Also, if there pgrep returns nothing, but `mega-cmd-server` complains about another instance, do launch it with:
```
mega-cmd-server --debug-full --skip-lock-check | tee megacmd.log
```

**Comment by ainquisitive** [comment]
Hi, doing killall -15 mega-cmd-server did nothing
pgrep isn't command on NAS systems (or at least ours)
doing:
mega-cmd-server --debug-full --skip-lock-check | tee megacmd.log
brings up the server, but megacmd.log is empty
